VMEncryption/main/EncryptionSettingsUtil.py

```diff
@@ -178,16 +181,31 @@ def controller_id_and_lun_to_settings_data(scsi_controller, lun_number):
                     }]
                 })
 
+        full_protector_path = os.path.join(CommonVariables.encryption_key_mount_point, protector_name)
+        with open(full_protector_path) as protector_file:
+            protector = protector_file.read()
+            protector_base64 = base64.standard_b64encode(protector)
+            protectors = [{"Name": protector_name, "Base64Key": protector_base64}]
+
+        protectors_name_only = [{"Name": protector["Name"], "Base64Key": "REDACTED"} for protector in protectors ]
+
         data = {
-            "DiskEncryptionDataVersion": self._CURRENT_DISK_ENCRYPTION_DATA_VERSION,
+            "DiskEncryptionDataVersion": self._DISK_ENCRYPTION_DATA_VERSION_V4,
             "DiskEncryptionOperation": "EnableEncryption",
             "KeyVaultUrl": kv_url,
             "KeyVaultResourceId": kv_id,
             "KekUrl": kek_url,
             "KekVaultResourceId": kek_kv_id,
             "KekAlgorithm": kek_algorithm,
+            "Protectors": protectors_name_only,
             "Disks": data_disks_settings_data
         }
+
+        self.logger.log("Settings without the protectors array: " + json.dumps(data, sort_keys=True, indent=4))
+        self.logger.log("Full Settings JSON might be found later in the BEK VOLUME")
```

Comment:
We need to remove dumping the settings.json file with the encryption passphrase key to bek volume. Passphrase in the settings.json file may go unoticed and also gives a wrong impression that it does not contain any secrets while we are dumping the key in there

Reply:
fixed